src/common/dav/tx.c

```diff
@@ -507,7 +507,25 @@ dav_tx_begin(dav_obj_t *pop, jmp_buf env, ...)
 
 		VALGRIND_START_TX;
 	} else if (tx->stage == DAV_TX_STAGE_NONE) {
+		struct umem_wal_tx *utx;
+
 		DAV_DBG("");
+		utx = pop->do_utx;
+		if (utx == NULL) {
```

Comment:
The 'do_utx' could be non-NULL if it was created by a reserve call before? I'm wondering if we should differentiate the two cases of redo from reserve and redo from tx, so that code will be less bug prone? Though that could introduce more complexities. Just a suggestion to consider, not necessary to make change in this PR.

Reply:
We will relook at this. Right now, this code is to handle case where we want to support certain atomic operation that does not want its change to be immediately committed but committed later along with next transaction that happens on this vos object.